web/src/app/components/login-page/login-page.component.ts

```diff
@@ -1,31 +1,60 @@
-import { Component, OnInit, ViewEncapsulation } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { AuthService } from '../../services/auth.service';
+import { UserService } from '../../services/user.service';
 
 @Component({
   selector: 'app-login-page',
   templateUrl: './login-page.component.html',
-  styleUrls: ['./login-page.component.scss'],
-  encapsulation: ViewEncapsulation.None,
+  styleUrls: ['./login-page.component.scss']
 })
-
 export class LoginPageComponent implements OnInit {
 
-  constructor(private authService: AuthService) { }
+  constructor(
+    private authService: AuthService,
+    private userService: UserService,
+  ) {}
 
-  ngOnInit() {
-  }
+  public hasError = false;
+  public errorMessage = '';
+
+  ngOnInit() {}
 
   loginWithGithub() {
     this.authService.signInWithGithub()
-      .subscribe((result) => {
-        console.log('result', result); // TODO: remove after firestore implementation
+      .subscribe((user) => {
+        if (!user) {
+          this.hasError = true;
+          this.errorMessage = 'Something went wrong. Please try again';
+          return;
+        }
+        if (user) {
+          this.saveUser(user, 'github');
+        }
       });
   }
 
   loginWithTwitter() {
     this.authService.signInWithTwitter()
-      .subscribe((result) => {
-        console.log('login with twitter', result); // TODO: remove after firestore implementation
+      .subscribe((user) => {
+        if (user) {
```

Comment:
This looks different to the `GitHub` login, there is no empty user check?

Reply:
I am working on testing the flow of twitter login.So I haven't yet handled the error scenario. Will do in the Twitter PR.

Reply:
Ok